<!doctype html><html lang=en class="js csstransforms3d"><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=generator content="Hugo 0.134.3"><meta name=description content><meta name=author content="thienlh@thienlu.com"><link rel=icon href=../../images/favicon.png type=image/png><title>Blog 1 :: Internship Report</title>
<link href=../../css/nucleus.css?1765261581 rel=stylesheet><link href=../../css/fontawesome-all.min.css?1765261581 rel=stylesheet><link href=../../css/hybrid.css?1765261581 rel=stylesheet><link href=../../css/featherlight.min.css?1765261581 rel=stylesheet><link href=../../css/perfect-scrollbar.min.css?1765261581 rel=stylesheet><link href=../../css/auto-complete.css?1765261581 rel=stylesheet><link href=../../css/atom-one-dark-reasonable.css?1765261581 rel=stylesheet><link href=../../css/theme.css?1765261581 rel=stylesheet><link href=../../css/hugo-theme.css?1765261581 rel=stylesheet><link href=../../css/theme-workshop.css?1765261581 rel=stylesheet><script src=../../js/jquery-3.3.1.min.js?1765261581></script><style>:root #header+#content>#left>#rlblock_left{display:none!important}</style></head><body data-url=../../3-blogstranslated/3.1-blog1/><nav id=sidebar class=showVisitedLinks><div id=header-wrapper><div id=header><a id=logo href=../../><svg id="Layer_1" data-name="Layer 1" viewBox="0 0 60 30" width="30%"><defs><style>.cls-1{fill:#fff}.cls-2{fill:#f90;fill-rule:evenodd}</style></defs><title>AWS-Logo_White-Color</title><path class="cls-1" d="M14.09 10.85a4.7 4.7.0 00.19 1.48 7.73 7.73.0 00.54 1.19.77.77.0 01.12.38.64.64.0 01-.32.49l-1 .7a.83.83.0 01-.44.15.69.69.0 01-.49-.23 3.8 3.8.0 01-.6-.77q-.25-.42-.51-1a6.14 6.14.0 01-4.89 2.3 4.54 4.54.0 01-3.32-1.19 4.27 4.27.0 01-1.22-3.2 4.28 4.28.0 011.46-3.4A6.06 6.06.0 017.69 6.46a12.47 12.47.0 011.76.13q.92.13 1.91.36V5.73a3.65 3.65.0 00-.79-2.66A3.81 3.81.0 007.86 2.3a7.71 7.71.0 00-1.79.22 12.78 12.78.0 00-1.79.57 4.55 4.55.0 01-.58.22h-.26q-.35.0-.35-.52V2a1.09 1.09.0 01.12-.58 1.2 1.2.0 01.47-.35A10.88 10.88.0 015.77.32 10.19 10.19.0 018.36.0a6 6 0 014.35 1.35 5.49 5.49.0 011.38 4.09zM7.34 13.38a5.36 5.36.0 001.72-.31A3.63 3.63.0 0010.63 12 2.62 2.62.0 0011.19 11a5.63 5.63.0 00.16-1.44v-.7a14.35 14.35.0 00-1.53-.28 12.37 12.37.0 00-1.56-.1 3.84 3.84.0 00-2.47.67A2.34 2.34.0 005 11a2.35 2.35.0 00.61 1.76A2.4 2.4.0 007.34 13.38zm13.35 1.8a1 1 0 01-.64-.16 1.3 1.3.0 01-.35-.65L15.81 1.51a3 3 0 01-.15-.67.36.36.0 01.41-.41H17.7a1 1 0 01.65.16 1.4 1.4.0 01.33.65l2.79 11 2.59-11A1.17 1.17.0 0124.39.6a1.1 1.1.0 01.67-.16H26.4a1.1 1.1.0 01.67.16 1.17 1.17.0 01.32.65L30 12.39 32.88 1.25A1.39 1.39.0 0133.22.6a1 1 0 01.65-.16h1.54a.36.36.0 01.41.41 1.36 1.36.0 010 .26 3.64 3.64.0 01-.12.41l-4 12.86a1.3 1.3.0 01-.35.65 1 1 0 01-.64.16H29.25a1 1 0 01-.67-.17 1.26 1.26.0 01-.32-.67L25.67 3.64l-2.56 10.7a1.26 1.26.0 01-.32.67 1 1 0 01-.67.17zm21.36.44a11.28 11.28.0 01-2.56-.29 7.44 7.44.0 01-1.92-.67 1 1 0 01-.61-.93v-.84q0-.52.38-.52a.9.9.0 01.31.06l.42.17a8.77 8.77.0 001.83.58 9.78 9.78.0 002 .2 4.48 4.48.0 002.43-.55 1.76 1.76.0 00.86-1.57 1.61 1.61.0 00-.45-1.16A4.29 4.29.0 0043 9.22l-2.41-.76A5.15 5.15.0 0138 6.78a3.94 3.94.0 01-.83-2.41 3.7 3.7.0 01.45-1.85 4.47 4.47.0 011.19-1.37 5.27 5.27.0 011.7-.86A7.4 7.4.0 0142.6.0a8.87 8.87.0 011.12.07q.57.07 1.08.19t.95.26a4.27 4.27.0 01.7.29 1.59 1.59.0 01.49.41.94.94.0 01.15.55v.79q0 .52-.38.52a1.76 1.76.0 01-.64-.2 7.74 7.74.0 00-3.2-.64 4.37 4.37.0 00-2.21.47 1.6 1.6.0 00-.79 1.48 1.58 1.58.0 00.49 1.18 4.94 4.94.0 001.83.92L44.55 7a5.08 5.08.0 012.57 1.6A3.76 3.76.0 0147.9 11a4.21 4.21.0 01-.44 1.93 4.4 4.4.0 01-1.21 1.47 5.43 5.43.0 01-1.85.93A8.25 8.25.0 0142.05 15.62z"/><path class="cls-2" d="M45.19 23.81C39.72 27.85 31.78 30 25 30A36.64 36.64.0 01.22 20.57c-.51-.46-.06-1.09.56-.74A49.78 49.78.0 0025.53 26.4 49.23 49.23.0 0044.4 22.53C45.32 22.14 46.1 23.14 45.19 23.81z"/><path class="cls-2" d="M47.47 21.21c-.7-.9-4.63-.42-6.39-.21-.53.06-.62-.4-.14-.74 3.13-2.2 8.27-1.57 8.86-.83s-.16 5.89-3.09 8.35c-.45.38-.88.18-.68-.32C46.69 25.8 48.17 22.11 47.47 21.21z"/></svg></a></div><div class=searchbox><label for=search-by><i class="fas fa-search"></i></label>
<input data-search-input id=search-by type=search placeholder=Search...>
<span data-search-clear><i class="fas fa-times"></i></span></div><script type=text/javascript src=../../js/lunr.min.js?1765261581></script><script type=text/javascript src=../../js/auto-complete.js?1765261581></script><script type=text/javascript>var baseurl="https://thienluhoan.github.io/workshop-template/"</script><script type=text/javascript src=../../js/search.js?1765261581></script></div><div class=highlightable><ul class=topics><li data-nav-id=/1-worklog/ title=Worklog class=dd-item><a href=../../1-worklog/><b>1. </b>Worklog
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/1-worklog/1.1-week1/ title="Week 1 Worklog" class=dd-item><a href=../../1-worklog/1.1-week1/><b>1.1. </b>Week 1 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.2-week2/ title="Week 2 Worklog" class=dd-item><a href=../../1-worklog/1.2-week2/><b>1.2. </b>Week 2 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.3-week3/ title="Week 3 Worklog" class=dd-item><a href=../../1-worklog/1.3-week3/><b>1.3. </b>Week 3 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.4-week4/ title="Week 4 Worklog" class=dd-item><a href=../../1-worklog/1.4-week4/><b>1.4. </b>Week 4 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.5-week5/ title="Week 5 Worklog" class=dd-item><a href=../../1-worklog/1.5-week5/><b>1.5. </b>Week 5 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.6-week6/ title="Week 6 Worklog" class=dd-item><a href=../../1-worklog/1.6-week6/><b>1.6. </b>Week 6 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.7-week7/ title="Week 7 Worklog" class=dd-item><a href=../../1-worklog/1.7-week7/><b>1.7. </b>Week 7 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.8-week8/ title="Week 8 Worklog" class=dd-item><a href=../../1-worklog/1.8-week8/><b>1.8. </b>Week 8 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.9-week9/ title="Week 9 Worklog" class=dd-item><a href=../../1-worklog/1.9-week9/><b>1.9. </b>Week 9 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.10-week10/ title="Week 10 Worklog" class=dd-item><a href=../../1-worklog/1.10-week10/><b>1.10. </b>Week 10 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.11-week11/ title="Week 11 Worklog" class=dd-item><a href=../../1-worklog/1.11-week11/><b>1.11. </b>Week 11 Worklog
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/1-worklog/1.12-week12/ title="Week 12 Worklog" class=dd-item><a href=../../1-worklog/1.12-week12/><b>1.12 </b>Week 12 Worklog
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/2-proposal/ title=Proposal class=dd-item><a href=../../2-proposal/><b>2. </b>Proposal
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/ title="Translated Blogs" class="dd-item
parent"><a href=../../3-blogstranslated/><b>3. </b>Translated Blogs
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/3-blogstranslated/3.1-blog1/ title="Blog 1" class="dd-item
active"><a href=../../3-blogstranslated/3.1-blog1/><b>3.1. </b>Blog 1
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/3.2-blog2/ title="Blog 2" class=dd-item><a href=../../3-blogstranslated/3.2-blog2/><b>3.2. </b>Blog 2
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/3.3-blog3/ title="Blog 3" class=dd-item><a href=../../3-blogstranslated/3.3-blog3/><b>3.3. </b>Blog 3
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/3.4-blog4/ title="Blog 4" class=dd-item><a href=../../3-blogstranslated/3.4-blog4/><b>3.4. </b>Blog 4
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/3.5-blog5/ title="Blog 5" class=dd-item><a href=../../3-blogstranslated/3.5-blog5/><b>3.5. </b>Blog 5
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/3-blogstranslated/3.6-blog6/ title="Blog 6" class=dd-item><a href=../../3-blogstranslated/3.6-blog6/><b>3.6. </b>Blog 6
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/4-eventparticipated/ title="Events Participated" class=dd-item><a href=../../4-eventparticipated/><b>4. </b>Events Participated
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/4-eventparticipated/4.1-event1/ title="Event 1" class=dd-item><a href=../../4-eventparticipated/4.1-event1/><b>4.1. </b>Event 1
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/4-eventparticipated/4.2-event2/ title="Event 2" class=dd-item><a href=../../4-eventparticipated/4.2-event2/><b>4.2. </b>Event 2
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/4-eventparticipated/4.3-event3/ title="Event 3" class=dd-item><a href=../../4-eventparticipated/4.3-event3/><b>4.3. </b>Event 3
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/4-eventparticipated/4.4-event4/ title="Event 4" class=dd-item><a href=../../4-eventparticipated/4.4-event4/><b>4.4. </b>Event 4
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/4-eventparticipated/4.5-event5/ title="Event 5" class=dd-item><a href=../../4-eventparticipated/4.5-event5/><b>4.5. </b>Event 5
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/5-workshop/ title=Workshop class=dd-item><a href=../../5-workshop/><b>5. </b>Workshop
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/5-workshop/5.1-workshop-overview/ title=Introduction class=dd-item><a href=../../5-workshop/5.1-workshop-overview/><b>5.1. </b>Introduction
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.2-prerequiste/ title=Prerequiste class=dd-item><a href=../../5-workshop/5.2-prerequiste/><b>5.2. </b>Prerequiste
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.3-s3-vpc/ title="Network Infrastructure Setup" class=dd-item><a href=../../5-workshop/5.3-s3-vpc/><b>5.3. </b>Network Infrastructure Setup
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/5-workshop/5.3-s3-vpc/5.3.1-create-gwe/ title="Create VPC & Subnets" class=dd-item><a href=../../5-workshop/5.3-s3-vpc/5.3.1-create-gwe/><b>5.3.1 </b>Create VPC & Subnets
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.3-s3-vpc/5.3.2-test-gwe/ title="Configure Internet & NAT Gateway" class=dd-item><a href=../../5-workshop/5.3-s3-vpc/5.3.2-test-gwe/><b>5.3.2 </b>Configure Internet & NAT Gateway
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/5-workshop/5.4-s3-onprem/ title="Data Layer Deployment" class=dd-item><a href=../../5-workshop/5.4-s3-onprem/><b>5.4. </b>Data Layer Deployment
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/5-workshop/5.4-s3-onprem/5.4.1-prepare/ title="Set up Security Groups" class=dd-item><a href=../../5-workshop/5.4-s3-onprem/5.4.1-prepare/><b>5.4.1 </b>Set up Security Groups
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.4-s3-onprem/5.4.2-create-interface-enpoint/ title="Initialize Amazon RDS" class=dd-item><a href=../../5-workshop/5.4-s3-onprem/5.4.2-create-interface-enpoint/><b>5.4.2 </b>Initialize Amazon RDS
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.4-s3-onprem/5.4.3-test-endpoint/ title="Initialize ElastiCache Redis" class=dd-item><a href=../../5-workshop/5.4-s3-onprem/5.4.3-test-endpoint/><b>5.4.3 </b>Initialize ElastiCache Redis
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/5-workshop/5.5-policy/ title="Application Deployment" class=dd-item><a href=../../5-workshop/5.5-policy/><b>5.5. </b>Application Deployment
<i class="fas fa-check read-icon"></i></a><ul><li data-nav-id=/5-workshop/5.5-policy/5.5.2-initialize-elastic-beanstalk/ title="Initialize Elastic Beanstalk" class=dd-item><a href=../../5-workshop/5.5-policy/5.5.2-initialize-elastic-beanstalk/><b>5.5.2 </b>Initialize Elastic Beanstalk
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.5-policy/5.5.1-package-with-docker/ title="Package with Docker" class=dd-item><a href=../../5-workshop/5.5-policy/5.5.1-package-with-docker/><b>5.5.1 </b>Package with Docker
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/5-workshop/5.6-monitoringoperations/ title="Monitoring & Operations" class=dd-item><a href=../../5-workshop/5.6-monitoringoperations/><b>5.6. </b>Monitoring & Operations
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/5-workshop/5.7-clean-up/ title="Clean Up" class=dd-item><a href=../../5-workshop/5.7-clean-up/><b>5.7. </b>Clean Up
<i class="fas fa-check read-icon"></i></a></li></ul></li><li data-nav-id=/6-self-evaluation/ title=Self-Assessment class=dd-item><a href=../../6-self-evaluation/><b>6. </b>Self-Assessment
<i class="fas fa-check read-icon"></i></a></li><li data-nav-id=/7-feedback/ title="Sharing and Feedback" class=dd-item><a href=../../7-feedback/><b>7. </b>Sharing and Feedback
<i class="fas fa-check read-icon"></i></a></li></ul><section id=shortcuts><h3>More</h3><ul><li><a class=padding href=https://www.facebook.com/groups/awsstudygroupfcj/><i class='fab fa-facebook'></i> AWS Study Group</a></li></ul></section><section id=prefooter><hr><ul><li><a class=padding><i class="fas fa-language fa-fw"></i><div class=select-style><select id=select-language onchange="location=this.value"><option id=en value=https://thienluhoan.github.io/workshop-template/3-blogstranslated/3.1-blog1/ selected>English</option><option id=vi value=https://thienluhoan.github.io/workshop-template/vi/3-blogstranslated/3.1-blog1/>Tiếng Việt</option></select><svg id="Capa_1" xmlns:xlink="http://www.w3.org/1999/xlink" width="255" height="255" viewBox="0 0 255 255" style="enable-background:new 0 0 255 255"><g><g id="arrow-drop-down"><polygon points="0,63.75 127.5,191.25 255,63.75"/></g></g></svg></div></a></li><li><a class=padding href=# data-clear-history-toggle><i class="fas fa-history fa-fw"></i> Clear History</a></li></ul></section><section id=footer><left><b>Workshop</b><br><img src="https://hitwebcounter.com/counter/counter.php?page=7920860&style=0038&nbdigits=9&type=page&initCount=0" title=Migrate alt="web counter" border=0></a><br><b><a href=https://cloudjourney.awsstudygroup.com/>Cloud Journey</a></b><br><img src="https://hitwebcounter.com/counter/counter.php?page=7830807&style=0038&nbdigits=9&type=page&initCount=0" title="Total CLoud Journey" alt="web counter" border=0>
</left><left><br><br><b>Last Updated</b><br><i><span id=lastUpdated style=color:orange></span>
</i><script>const today=new Date,formattedDate=today.toLocaleDateString("en-GB");document.getElementById("lastUpdated").textContent=formattedDate</script></left><left><br><br><b>Team</b><br><i><a href=https://www.facebook.com/groups/660548818043427 style=color:orange>First Cloud Journey</a><br></i></left><script async defer src=https://buttons.github.io/buttons.js></script></section></div></nav><section id=body><div id=overlay></div><div class="padding highlightable"><div><div id=top-bar><div id=breadcrumbs itemscope itemtype=http://data-vocabulary.org/Breadcrumb><span id=sidebar-toggle-span><a href=# id=sidebar-toggle data-sidebar-toggle><i class="fas fa-bars"></i>
</a></span><span id=toc-menu><i class="fas fa-list-alt"></i></span>
<span class=links><a href=../../>Internship Report</a> > <a href=../../3-blogstranslated/>Translated Blogs</a> > Blog 1</span></div><div class=progress><div class=wrapper><nav id=TableOfContents><ul><li><a href=#protein-structure-prediction--overview>Protein Structure Prediction – Overview</a></li><li><a href=#introducing-openfold3>Introducing OpenFold3</a></li><li><a href=#first-challenge-an-optimal-research-environment>First Challenge: An Optimal Research Environment</a></li><li><a href=#second-challenge-the-msa-generation-process>Second Challenge: The MSA Generation Process</a></li><li><a href=#third-challenge-aiml-training>Third Challenge: AI/ML Training</a><ul><li><a href=#gpu-resource-allocation>GPU Resource Allocation:</a></li><li><a href=#cost-optimization>Cost Optimization:</a></li><li><a href=#distributed-training-infrastructure>Distributed Training Infrastructure:</a></li></ul></li><li><a href=#conclusion>Conclusion</a></li><li><a href=#about-the-authors>About the Authors</a><ul><li><a href=#daniele-granata>Daniele Granata</a></li><li><a href=#anamaria-todor>Anamaria Todor</a></li><li><a href=#arthur-grabman>Arthur Grabman</a></li><li><a href=#rômulo-jales>Rômulo Jales</a></li><li><a href=#jacob-mevorach>Jacob Mevorach</a></li><li><a href=#stig-bøgelund-nielsen>Stig Bøgelund Nielsen</a></li></ul></li></ul></nav></div></div></div></div><div id=head-tags></div><div id=body-inner><h1>Blog 1</h1><h1 id=how-novo-nordisk-columbia-university-and-aws-collaborated-with-the-openfold-ai-consortium-to-develop-openfold3>How Novo Nordisk, Columbia University and AWS collaborated with the OpenFold AI Consortium to develop OpenFold3</h1><p>by Daniele Granata, Anamaria Todor, Arthur Grabman, Rômulo Jales, Jacob Mevorach, and Stig Bøgelund Nielsen | on 03 JUN 2025 in | Customer Solution | High Performance Computing | Life Sciences Permalink</p><p>This post was contributed by Daniele Granata, Principal Modelling Scientist; Søren Moos, Architect Advisor Director; Rômulo Jales, Senior Software Engineer; Stig Bøgelund Nielsen, Senior HPC Cloud Engineer at Novo Nordisk; and Jake Mevorach, Senior Specialist, Application Modernization; Arthur Grabman, Principal Technical Account Manager; and Anamaria Todor, Principal Solutions Architect at AWS.</p><p>In this blog post, we are excited to share how Novo Nordisk, Columbia University, and AWS collaborated to develop OpenFold3, an advanced protein structure prediction model, using scalable and cost-effective bioinformatics solutions on AWS.</p><p>The OpenFold Consortium, a non-profit AI research and development group, has been at the forefront of creating open-source tools for biology and drug development. Their latest project, OpenFold3, represents a significant advancement in protein structure prediction. This post will explore how we leveraged AWS services and innovative approaches to overcome challenges in training this complex model while maintaining cost efficiency and scalability.</p><p>First, we will introduce protein structure prediction and discuss its importance in advancing medical research and pharmaceutical development.</p><p>Next, we will dive into three main challenges we faced and how we addressed them:</p><ol><li><strong>Creating a lightweight research environment:</strong></li></ol><p>We will discuss how we developed the Research Collaboration Platform (RCP) using AWS services, providing a secure, flexible, and efficient workspace for collaborative efforts.</p><ol start=2><li><strong>Creating Multiple Sequence Alignments (MSAs):</strong></li></ol><p>We will explore how we optimized the MSA generation process using AWS Graviton processors, achieving significant improvements in both speed and cost-efficiency.</p><ol start=3><li><strong>AI/ML Training:</strong></li></ol><p>We will present our approach to the computationally intensive task of training the OpenFold3 model, including the use of Amazon EC2 Capacity Blocks for ML and Spot Instances to efficiently manage resources.</p><p>This post provides valuable insights for researchers, data scientists, and organizations looking to leverage cloud computing for complex bioinformatics tasks. Whether you&rsquo;re working on protein structure prediction or other computationally intensive projects in life sciences, the strategies and solutions discussed can help you optimize workflows for both performance and cost-effectiveness.</p><h2 id=protein-structure-prediction--overview>Protein Structure Prediction – Overview</h2><p>If you&rsquo;re interested in curing diseases, advancing medicine, and even saving lives, a great place to start is understanding protein structures. There are many different types of proteins in the human body. Some proteins are beneficial, such as the lactase enzyme, which helps digest dairy products. Other proteins can be harmful and lead to disease or even death. Even small changes in the structure of these microscopic compounds can determine survival.</p><p>Protein-based therapeutics is a new and promising approach to treating diseases. However, because proteins have a strong impact on the human body, we need to ensure that we create proteins with the appropriate structure to treat diseases.</p><p>The traditional process of developing protein-based therapeutics typically includes:</p><ol><li>Finding receptors or other structures in the body (in the industry called &ldquo;targets&rdquo;) that you believe a protein can bind to achieve the desired therapeutic effect.</li><li>Developing a process to produce a protein capable of acting on this target.</li><li>Verifying through methods such as X-ray crystallography or Nuclear Magnetic Resonance (NMR) Spectroscopy that the production process creates a protein with the desired structure.</li><li>Conducting extensive research and testing, ultimately clinical trials, to ensure the newly developed protein achieves the desired effect.</li></ol><p>Historically, the biggest problem was that while we could genetically engineer proteins with specific sequences, we lacked good tools to predict protein structure from genetic data. This made the therapeutic development process time-consuming, because even when you know how to create a protein with a specific genetic sequence, you still need many highly trained personnel operating expensive equipment—even by global pharmaceutical company standards—to verify the structure. If the structure is incorrect, you must start over and repeat the process until the correct structure is achieved. The process must be restarted each time a new molecule needs to be characterized.</p><p>All of this makes developing protein-based therapeutics extremely expensive and time-consuming. A breakthrough emerged when researchers realized that AI could be applied to significantly accelerate this process. Researchers had large amounts of historical data, combining protein structures with corresponding genetic data. The question was posed:</p><p>&ldquo;If we have data combining protein structures with genetic data, can we use it to train a model that, when provided with a new genetic sequence, will predict the corresponding protein structure?&rdquo;</p><p>The research community answered &ldquo;yes,&rdquo; and from that point, many AI models emerged that take genetic sequences as input and output predicted protein structures.</p><p>Running these models requires only a small amount of electrical power but provides crucial feedback about protein behavior before starting expensive and time-consuming experiments.</p><h2 id=introducing-openfold3>Introducing OpenFold3</h2><p>The OpenFold Consortium is a non-profit AI research and development consortium, specializing in developing free open-source tools to support biology and drug discovery. One of the tools they develop is OpenFold, an AI model that takes genetic sequences as input and outputs predicted protein structures.</p><p>OpenFold3 is the latest version of this software, and to accelerate its development, Novo Nordisk and AWS collaborated with the OpenFold Consortium and Columbia University.</p><h2 id=first-challenge-an-optimal-research-environment>First Challenge: An Optimal Research Environment</h2><p>We started this project with limited time and budget, and the first challenge the project faced was that researchers needed a lightweight research environment as soon as possible.</p><p>As the priority scientific partner, Novo Nordisk needed to create an environment that could fundamentally change how research collaboration works. This environment had to enable seamless collaboration between Novo Nordisk, Columbia University, and the OpenFold Consortium, while strictly adhering to security and compliance standards.</p><p>It was important that researchers could use their preferred tools and devices, maintaining established workflows without disruption. With a tight timeline, we needed rapid setup capability and short infrastructure build times, combined with a streamlined onboarding process for external collaborators.</p><p>The system also needed to be flexible enough to meet different consortium contract terms, particularly regarding geographic data location and access controls. Most importantly, this environment had to accelerate early target discovery by providing robust computational resources while maintaining cost efficiency. Finally, an environment was needed that could dynamically scale up during intensive training phases but could scale down when not in use to maintain cost efficiency.</p><p>All of these requirements had to be balanced with empowering researchers with creative freedom—a factor particularly important when considering the high computational requirements for training large language models for protein structure prediction.</p><p><img alt="Figure 1 – Research Collaboration Platform Architecture" src=../../images/3-BlogsTranslated/3.1-Blog1/rcp-architecture.png></p><p><em>Figure 1 – This diagram illustrates Novo Nordisk&rsquo;s Research Collaboration Platform (RCP), showing how the platform creates a secure and high-performance research environment on AWS Cloud. Key elements to note are: (1) Secure access layer using Okta authentication, allowing researchers to connect securely via web portals or SSH.</em></p><p><em>(2) Flexible computing resources managed by AWS ParallelCluster, automatically scaling to match researcher needs. (3) AWS EFS and AWS FSx file systems handle research data. The key point is that this architecture allows scientists to conduct complex research securely and efficiently, especially for high-demand tasks like AI model training, while ensuring regulatory compliance.</em></p><p><em>This platform stands out by combining enterprise-grade security with the flexibility researchers need, and can be deployed in less than 2 hours through automation.</em></p><p>To meet these requirements, Novo Nordisk developed the Research Collaboration Platform (RCP), a comprehensive solution built on AWS, providing a secure and flexible environment for computational research. The architecture (as illustrated in Figure 1) uses AWS ParallelCluster as the foundation, operating within a Virtual Private Cloud (VPC) to ensure secure isolation for research workloads.</p><p>At the center of the RCP security model is integration with Okta, providing robust user directory services and multi-factor authentication. Researchers can access the platform through web portals or SSH using Okta Advanced Server Access, ensuring secure and controlled access to resources. The platform entry point is an EC2 head node located in a public subnet, containing necessary research tools including RStudio, Docker containers, and scientific frameworks like Singularity.</p><p>The compute infrastructure is intelligently designed with SLURM job scheduler, managing workloads across different instance types, optimized for diverse computational needs—from general-purpose compute to memory-intensive tasks and GPU acceleration. Amazon EC2 Auto Scaling ensures resources are allocated dynamically based on demand, helping maintain cost efficiency while meeting computational requirements.</p><p>The platform uses Amazon Aurora Serverless for database management and Amazon DynamoDB for cluster state information, providing reliable and scalable data storage solutions.</p><p>Data management is handled through a combination of Amazon EFS and FSx for Lustre, providing both general-purpose and high-performance file storage. The entire infrastructure is monitored by Amazon CloudWatch, with AWS CloudFormation enabling infrastructure as code, ensuring consistent and repeatable deployments. Deploying an RCP instance takes less than 2 hours.</p><p>This architecture provides researchers with flexibility to use their preferred tools while ensuring security and compliance requirements. The combination of containerization, automated scaling, and high-performance computing capabilities enables rapid experimentation and accelerates research workflows, especially for computationally intensive tasks like training the OpenFold3 model.</p><h2 id=second-challenge-the-msa-generation-process>Second Challenge: The MSA Generation Process</h2><p>With the research environment deployed and operational, we had to find a way to generate millions of Multiple Sequence Alignments (MSAs). A multiple sequence alignment (MSA) is essentially a unit of information describing the degree of similarity between three or more genetic sequences. This is important because these MSAs (along with other data) will be used to train OpenFold3.</p><p>The software we were tasked with scaling to generate these MSAs is HHBlits (specifically version 3, available for free on GitHub). When calculating cost and runtime for this step, we found it would take twice the time and more than double the cost compared to the expected budget and timeline if using the initial compute instances, r5.16xlarge.</p><p>Amazon EC2 R8g instances, running on the latest generation AWS Graviton4 processors, had the potential to provide better price performance for this memory-intensive workload. Benchmark testing on r8g.16xlarge showed 50% lower runtime and 55% lower cost for HHBlits compared to r5.16xlarge.</p><p>Combined with some other optimizations, the entire process ran much faster at lower cost, allowing us to generate over one million MSAs per day.</p><p>One challenge when transitioning from R5 to R8g was the difference in CPU architecture. R8g uses ARM-based CPUs, while R5 is x86-64. In terms of source code, this was not an issue because all code is C, C++, and Python, supporting both AArch64 and x86-64. However, we had to create an entirely new cluster because ParallelCluster requires head nodes and compute nodes to have the same CPU type.</p><p>We solved this by defining a new AWS ParallelCluster configuration, synchronizing CPU architecture between head nodes and compute nodes. To improve performance, we also customized the EC2 image with EBS as local storage. All generated MSAs were stored directly into an Amazon S3 bucket immediately. This configuration proved to be very efficient.</p><p>From the end-user perspective, there was very little change. Scientists only needed to use a different host name to log in and access the Graviton Slurm queues. We also reused the same EFS and FSx for Lustre filesystems configured initially without performance penalties.</p><p>The best aspect of this architecture is its high scalability. If we want to adjust the number of MSAs generated simultaneously, we simply add or remove nodes. The figure of one million MSAs mentioned above is not the maximum limit; in fact, we don&rsquo;t yet know the upper limit for throughput when using this method.</p><h2 id=third-challenge-aiml-training>Third Challenge: AI/ML Training</h2><p>As is known, the key factor for running AI/ML workloads is GPUs. But for protein prediction, GPUs alone are not enough. The system must also meet a range of prerequisites including petabyte-scale storage, high-speed networking, and hundreds of GPUs.</p><p>The OpenFold3 project required up to 256 GPUs for training. This equates to 32 p5en.48xlarge EC2 instances running in parallel, processing and generating data over several weeks, 24 hours a day.</p><p>This configuration requires infrastructure that is scalable, resilient, observable, and most importantly, flexible.</p><p>Initially, scientists needed to build training logic almost from scratch. The level of uncertainty was high; therefore, 32 GPU machines were not necessary from the start. What they needed was an environment to develop, conduct small experiments, and gradually scale up the number of nodes used for training. This is exactly the infrastructure capability that AWS ParallelCluster can provide.</p><p>When training OpenFold3 on the generated data, we faced three main challenges:</p><ol><li>Budget constraints required a cost-effective solution, avoiding costly errors.</li><li>Increasing model complexity required distributed AI/ML training infrastructure, as the model no longer fits on a single GPU or instance.</li><li>Securing large GPU resources for both training and preliminary testing required quick action and coordination among all stakeholders.</li></ol><p>To address these challenges, we implemented the following solutions:</p><h3 id=gpu-resource-allocation>GPU Resource Allocation:</h3><p>We used Amazon EC2 Capacity Blocks for ML, a consumption model that allows reservation of high-performance GPUs for short periods for machine learning workloads. This service allows users to reserve hundreds of NVIDIA GPUs synchronously in Amazon EC2 UltraClusters, specifying cluster size, future start date, and duration. Capacity Blocks provide reliable, predictable GPU access without long-term commitments, ideal for training and fine-tuning ML models, running experiments, and preparing for future scaling needs. Pricing is dynamic, based on supply and demand, typically around P5 On-Demand rates.</p><h3 id=cost-optimization>Cost Optimization:</h3><p>To optimize costs, we used Amazon EC2 Spot instances. Spot instances allow use of spare EC2 capacity with discounts up to 90% compared to On-Demand pricing for fault-tolerant and flexible workloads. As AWS GPU infrastructure expands, many new GPU instance types are available through the Spot pricing model, enabling large-scale workloads to run at significantly lower costs thanks to AWS&rsquo;s massive operational scale. Spot instances can be reclaimed by AWS when EC2 needs resources, typically with a two-minute interruption notice, helping applications handle transitions gracefully.</p><p>By implementing these methods, Novo Nordisk&rsquo;s RCP team achieved an 85% cost reduction compared to using On-Demand instances for our use case.</p><h3 id=distributed-training-infrastructure>Distributed Training Infrastructure:</h3><p>We drew inspiration from AWSome Distributed Training examples, particularly those about DeepSpeed, to build our scalable training infrastructure. This configuration efficiently uses GPUs from multiple instances simultaneously, creating a highly effective and scalable system. The final infrastructure allows Slurm job submissions to an autoscaling, efficient, distributed GPU cluster, capable of handling the complex task of training the new OpenFold3 model.</p><p>By implementing these solutions, we successfully overcame the initial challenges, creating a cost-effective, scalable, and efficient training environment for OpenFold3.</p><h2 id=conclusion>Conclusion</h2><p>The successful training of OpenFold3 represents an important collaboration between Novo Nordisk, Columbia University, and AWS, overcoming three major challenges through innovative solutions:</p><ol><li><p>Novo Nordisk developed the Research Collaboration Platform (RCP), a secure and flexible research environment built on AWS, enabling seamless collaboration while maintaining strict security standards. The platform can be deployed in less than two hours and provides dynamic scaling for computational resources.</p></li><li><p>Together, we optimized the Multiple Sequence Alignment (MSA) generation process using AWS Graviton processors, achieving a 50% reduction in runtime and a 55% cost reduction. This enabled researchers to generate over one million MSAs per day with a highly scalable architecture.</p></li><li><p>Together, we addressed AI/ML training challenges by combining Amazon EC2 Capacity Blocks for ML and Spot instances. Our distributed training infrastructure efficiently utilized 256 GPUs across 32 p5en.48xlarge EC2 machines.</p></li></ol><p>These solutions not only made the project cost-effective and efficient but also created a blueprint for future large-scale bioinformatics collaborations between Novo Nordisk and AWS.</p><p>The successful development of OpenFold3 advances the field of protein structure prediction, contributing to faster and more efficient therapeutic development processes.</p><p><strong>TAGS:</strong> Protein Folding</p><hr><h2 id=about-the-authors>About the Authors</h2><p><img alt="Daniele Granata" src=../../images/3-BlogsTranslated/3.1-Blog1/Daniele-Granata.png></p><h3 id=daniele-granata>Daniele Granata</h3><p>Daniele Granata is a Principal Modelling Scientist in the In Silico Protein Discovery field at Novo Nordisk. He is passionate about building connections between people and technology and working at the intersection of protein science, AI, and HPC. He holds a bachelor&rsquo;s and master&rsquo;s degree in Physics, a PhD in &ldquo;Physics and Chemistry of Biological Systems,&rdquo; and 5 years of postdoctoral experience in the US and Denmark.</p><p>Over the past 6 years at Novo Nordisk, Daniele has been prominent in advancing data science and generative AI to design new drugs for patients, while maintaining a passion for open sourcing for the scientific community.</p><p><img alt="Anamaria Todor" src=../../images/3-BlogsTranslated/3.1-Blog1/Anamaria-Todor.png></p><h3 id=anamaria-todor>Anamaria Todor</h3><p>Anamaria Todor is a Principal Solutions Architect based in Copenhagen, Denmark. She first encountered computers at the age of 4 and has never left computer science, video games, and engineering since. She has worked in various technical roles, from freelancer, full-stack developer, data engineer, technical lead, to CTO, at various companies in Denmark, focusing on the gaming and advertising industries. She has worked at AWS for over 5 years as a Principal Solutions Architect, primarily focusing on life sciences and AI/ML. Anamaria holds a bachelor&rsquo;s degree in Applied Engineering and Computer Science, a master&rsquo;s in Computer Science, and over 10 years of experience with AWS. When not working or playing video games, she coaches girls and women professionals to help them understand and find their path in the technology field.</p><p><img alt="Arthur Grabman" src=../../images/3-BlogsTranslated/3.1-Blog1/Arthur-Grabman.png></p><h3 id=arthur-grabman>Arthur Grabman</h3><p>Arthur Grabman is a Principal Technical Account Manager with over 15 years of experience, specializing in HPC for enterprise clients. He is passionate about translating complex technical concepts into concrete business value. Outside of work, Arthur enjoys traveling with his family.</p><p><img alt="Rômulo Jales" src=../../images/3-BlogsTranslated/3.1-Blog1/Romulo-Jales.png></p><h3 id=rômulo-jales>Rômulo Jales</h3><p>Rômulo Jales is a Senior Software Engineer at Novo Nordisk. With a solid foundation of over 15 years in software development, Rômulo enjoys completing work by creating simple and elegant solutions, even in critical mission applications.</p><p>Rômulo holds a bachelor&rsquo;s degree in Computer Engineering and is still driven by the initial curiosity about how and why things work the way they do. For him, there is always something new to learn and discover.</p><p><img alt="Jacob Mevorach" src=../../images/3-BlogsTranslated/3.1-Blog1/Jacob-Mevorach.png></p><h3 id=jacob-mevorach>Jacob Mevorach</h3><p>Jacob Mevorach is a Senior Specialist in containers for healthcare and life sciences at AWS. Jacob has a background in bioinformatics and machine learning. Before joining AWS, Jacob focused on supporting and implementing large-scale analyses for genomics and other scientific fields.</p><p><img alt="Stig Bøgelund Nielsen" src=../../images/3-BlogsTranslated/3.1-Blog1/Stig-Bogelund-Nielsen.png></p><h3 id=stig-bøgelund-nielsen>Stig Bøgelund Nielsen</h3><p>Stig Bøgelund Nielsen is a Senior Cloud & HPC Engineer at Novo Nordisk. A lifelong engagement with IT, starting from supporting local businesses, has shaped his career. Over 8 years at IBM provided a solid foundation, with roles such as IT Service and Infrastructure Architect, Team Lead, and Transition & Transformation Architect/Project Manager, all focused on creating business value through effective IT strategies and implementations.</p><p>He joined Novo Nordisk nearly two years ago as a Cloud & HPC Engineer and is currently the Tech Lead of the Research Collaboration Platform (RCP), directly supporting pioneering research, including the OpenFold project. He is passionate about leveraging technology to drive research and innovation.</p><footer class=footline></footer></div></div><div id=navigation><a class="nav nav-prev" href=../../3-blogstranslated/ title="Translated Blogs"><i class="fa fa-chevron-left"></i></a>
<a class="nav nav-next" href=../../3-blogstranslated/3.2-blog2/ title="Blog 2" style=margin-right:0><i class="fa fa-chevron-right"></i></a></div></section><div style=left:-1000px;overflow:scroll;position:absolute;top:-1000px;border:none;box-sizing:content-box;height:200px;margin:0;padding:0;width:200px><div style=border:none;box-sizing:content-box;height:200px;margin:0;padding:0;width:200px></div></div><script src=../../js/clipboard.min.js?1765261581></script><script src=../../js/perfect-scrollbar.min.js?1765261581></script><script src=../../js/perfect-scrollbar.jquery.min.js?1765261581></script><script src=../../js/jquery.sticky.js?1765261581></script><script src=../../js/featherlight.min.js?1765261581></script><script src=../../js/highlight.pack.js?1765261581></script><script>hljs.initHighlightingOnLoad()</script><script src=../../js/modernizr.custom-3.6.0.js?1765261581></script><script src=../../js/learn.js?1765261581></script><script src=../../js/hugo-learn.js?1765261581></script><link href=../../mermaid/mermaid.css?1765261581 rel=stylesheet><script src=../../mermaid/mermaid.js?1765261581></script><script>mermaid.initialize({startOnLoad:!0})</script><script>(function(e,t,n,s,o,i,a){e.GoogleAnalyticsObject=o,(e[o]=e[o]||function(){(e[o].q=e[o].q||[]).push(arguments)},e[o].l=1*new Date),(i=t.createElement(n),a=t.getElementsByTagName(n)[0]),i.async=1,i.src=s,a.parentNode.insertBefore(i,a)})(window,document,"script","https://www.google-analytics.com/analytics.js","ga"),ga("create","UA-158079754-2","auto"),ga("send","pageview")</script></body></html>